software/react-native/concepts.md

```diff
@@ -0,0 +1,60 @@
+# Basic Concepts
+
+#### Request / Response Data Flow
+
+In a standard redux project data flows like this:
+
+1. Dispatch an `attempting` action that will attempt to `POST`/`GET` data from a server. This action will usually flip an `attempting: true` flag in it's domain-related reducer in case we want to display some sort of UI to demonstrate to the user that a request is in process.
+
+2. A saga is listens for this `attempting` action, and then runs when it receives it along with it's params (*it will have passed through the reducer first*). The saga handles all side-effects associated with this action. The saga will be the place where the api call is actually made. Depending on it's success or failure a `success` or `failure` action will be dispatched to the reducer. If it's a `success` we dispatch the received data as well, if it's a `failure`, we display the received error messages.
+
+3. After the `success` action is dispatched from the reducer, the data will get stored as is or [normalized](https://github.com/paularmstrong/normalizr) and stored. We try to normalize all deeply nested data to make it much easier to work with. The `success` or `failure` action will also flip that `attempting: true` flag mentioned above back to `attempting: false` to show the user that the request has completed.
+
+4. Now that we have attempted to fetch our data, successfully received our data, and stored (or normalized) it, depending on how we are using it, we may need to *transform* the data before displaying or performing calculations with it.
+In a perfect world, we would not need to do many transformations since the API would return to us the data exactly as we needed *cough GraphQL*. However, often transforms are necessary to coerce the data into a more workable form. For example, List Views in React Native require very specific data structures, and we often have to transform our data to fit a List View's requirements.
+
```

Comment:
I think something that would be useful here is to implement the transforms/test that, but also wrap it with a `selector` (or simply let the selector do the transform itself, depending on how complex things are)

[reselect] (https://github.com/reactjs/reselect) is a great library for this

what this also does is make our transforms way more efficient since they are memoized and will not have to be computed multiple times upon component-rerender if thats happening a lot (as it usually does with React)